### PR TITLE
PCA, build and namespace Fixes

### PR DESCRIPTION
**Pull request by geoffreychiou**
@akgoins Please review.

1) Fixed some problems to allow for building with catkin tools.
2) @akgoins Fixed some bugs with PCA and added unit tests in sphere_discretization.cpp
3) Removed the use of using namespace to make the code easier to debug.

### CODE REVIEW

**Comment by jontromanab** [comment]
@akgoins  and @geoffreychiou I am working on the #11 
Is anybody else is working on that? Should I continue or wait for your updates?

**Comment by geoffreychiou** [comment]
@jontromanab I have not started working on it yet, if you have already started please continue, I can find something else.

- Geoffrey

**Comment by jontromanab** [comment]
@geoffreychiou I am currently working on #11 #30 and #16 
Hope you understand that the time limit was really small. That's why the codes are all messed up. Now its time to make them better.
You are doing an wonderful job. Keep up the good work.

**Comment by akgoins** [comment]
@geoffreychiou, Sorry, but merging the other PR has broken this one.  Please resolve conflict and I will review as soon as possible.

**Comment by geoffreychiou** [comment]
@akgoins I have resolved the conflict, please check whenever you have some free time. 
- Geoff

File: map_creator/CMakeLists.txt [hunk 1]
```diff
@@ -107,6 +107,6 @@ add_executable(load_capability_map src/load_capability_map.cpp)
 add_dependencies(load_capability_map ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 target_link_libraries(load_capability_map -lhdf5 -lhdf5_cpp hdf5_dataset ${catkin_LIBRARIES})
 
-
-
+catkin_add_gtest(utest test/utest.cpp)
+target_link_libraries(utest sphere_discretization ${catkin_LIBRARIES})
```
Comment (Levi-Armstrong) [inline map_creator/CMakeLists.txt]: I believe the catkin tests should be enclosed with the if statement below so the the gtest does not build every time.

```
if(CATKIN_ENABLE_TESTING)
  catkin_add_gtest(utest test/utest.cpp)
  target_link_libraries(utest sphere_discretization ${catkin_LIBRARIES})
endif()
```

**Comment by akgoins** [comment]
@geoffreychiou, if you could make the two changes that Levi mentioned, I will merge this PR.

File: map_creator/src/create_reachability_map.cpp [hunk 2]
```diff
@@ -47,10 +38,10 @@ int main(int argc, char **argv)
...
   time(&startit);
   float resolution = 0.08;
-  Kinematics k;
-  string ext = ".h5";
-  string filename =
-      string(k.getRobotName()) + "_" + "r" + str(boost::format("%d") % resolution) + "_" + "reachability" + "." + "h5";
+  kinematics::Kinematics k;
+  std::string ext = ".h5";
+  std::string filename =
+      std::string(k.getRobotName()) + "_" + "r" + str(boost::format("%d") % resolution) + "_" + "reachability" + "." + "h5";
```
Comment (geoffreychiou) [inline map_creator/src/create_reachability_map.cpp]: I had to move the ')' inside to get it to build.
```
str(boost::format("%s_r%d_reachability.h5") % k.getRobotName() % resolution);
```
Thanks for the suggestions.

**Comment by geoffreychiou** [comment]
@akgoins, I have added the changes Levi suggested.

**Comment by akgoins** [comment]
+1

**Pull request merged**